### PR TITLE
libnet: Remove unused cmd/readme_test

### PR DESCRIPTION
**Pull request by akerouanton**
**- What I did**

This command was originally added by https://github.com/moby/libnetwork/commit/ea7f555446401de09fc2410aba691bf70bbb2dee to test the code snippet put into libnet's README.md. Nothing compiles this file and it doesn't add any value to the project. So better remove it than maintaining it.

This commit also removes the code snippet from libnet's README.md for the same reasons.

### CODE REVIEW

File: libnetwork/README.md [hunk 1]
```diff
@@ -11,79 +11,6 @@ Please refer to the [design](docs/design.md) for more information.
 
 There are many networking solutions available to suit a broad range of use-cases. libnetwork uses a driver / plugin model to support all of these solutions while abstracting the complexity of the driver implementations by exposing a simple and consistent Network Model to users.
 
-
-```go
-package main
```
Comment (thaJeztah) [inline libnetwork/README.md:16]: Do you think the example itself is good to have in godoc? If we think there's still a use for it, we could consider an _actual_ "godoc" example, similar to https://github.com/moby/moby/blob/199793350807908c04b56741b11e6b943158ac51/api/types/filters/example_test.go#L4-L35

Which shows in the docs like https://pkg.go.dev/github.com/docker/docker/api/types/filters#example-Args.MatchKVList

**Review by thaJeztah — APPROVED** [review]
It looks like `libnetwork` itself doesn't have a proper GoDoc for the package description. As a result, it's currently using the package description for the generated protos. We could consider adding a `doc.go` and copying the relevant information from the README into it (at least the basic description); https://pkg.go.dev/github.com/docker/docker@v24.0.7+incompatible

Screenshot 2023-12-17 at 13 15 30

File: libnetwork/README.md (hunk 1 above)
Comment (akerouanton) [reply libnetwork/README.md:16]: Although libnetwork isn't an internal package, we don't expect any other project to use it. So I think it's not worth re-introducing this snippet as a doc example (although I agree we should fix the pkg description).

Comment (thaJeztah) [reply libnetwork/README.md:16]: Yeah, makes sense. Thought I'd mention it as an option before we rip it out, but agreed that it's probably not worth the effort to write new examples.

**Pull request merged**